### PR TITLE
Merge `github/Organization.pyi` back to source

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

File: github/Organization.py [hunk 1]
```diff
@@ -39,351 +39,290 @@
...
-    def add_to_members(self, member, role=github.GithubObject.NotSet):
+    def add_to_members(self, member: NamedUser, role: Opt[str] = NotSet) -> None:
         """
         :calls: `PUT /orgs/{org}/memberships/{user} <https://docs.github.com/en/rest/reference/orgs#update-an-organization-membership-for-the-authenticated-user>`_
-        :param member: :class:`github.NamedUser.NamedUser`
-        :param role: string
-        :rtype: None
         """
-        assert role is github.GithubObject.NotSet or isinstance(role, str), role
+        assert role is NotSet or isinstance(role, str), role
```
Comment (EnricoMi) [inline github/Organization.py:314]:
```suggestion
        assert is_optional(role, str), role
```

File: github/Organization.py [hunk 2]
```diff
@@ -39,351 +39,290 @@
...
         assert isinstance(member, github.NamedUser.NamedUser), member
         put_parameters = {}
-        if role is not github.GithubObject.NotSet:
+        if role is not NotSet:
```
Comment (EnricoMi) [inline github/Organization.py:317]:
```suggestion
        if is_defined(role):
```

File: github/Organization.py [hunk 3]
```diff
@@ -412,30 +347,25 @@ def create_fork(
...
-        :param private: bool
-        :rtype: :class:`github.Repository.Repository`
         """
         assert isinstance(name, str), name
         assert isinstance(repo, github.Repository.Repository), repo
-        assert description is github.GithubObject.NotSet or isinstance(description, str), description
-        assert private is github.GithubObject.NotSet or isinstance(private, bool), private
-        post_parameters = {
+        assert description is NotSet or isinstance(description, str), description
+        assert private is NotSet or isinstance(private, bool), private
```
Comment (EnricoMi) [inline github/Organization.py:361]:
```suggestion
        assert is_optional(description, str), description
        assert is_optional(private, bool), private
```

File: github/Organization.py [hunk 4]
```diff
@@ -462,30 +392,27 @@ def create_hook(
...
+        post_parameters: dict[str, Any] = {
             "name": name,
             "config": config,
         }
-        if events is not github.GithubObject.NotSet:
+        if events is not NotSet:
             post_parameters["events"] = events
-        if active is not github.GithubObject.NotSet:
+        if active is not NotSet:
             post_parameters["active"] = active
```
Comment (EnricoMi) [inline github/Organization.py:404]: This should use `NotSet.remove_unset_items`.

File: github/Organization.py [hunk 5]
```diff
@@ -497,112 +424,82 @@ def create_project(self, name, body=github.GithubObject.NotSet):
...
+        assert has_projects is NotSet or isinstance(has_projects, bool), has_projects
+        assert team_id is NotSet or isinstance(team_id, int), team_id
+        assert auto_init is NotSet or isinstance(auto_init, bool), auto_init
+        assert license_template is NotSet or isinstance(license_template, str), license_template
+        assert gitignore_template is NotSet or isinstance(gitignore_template, str), gitignore_template
+        assert allow_squash_merge is NotSet or isinstance(allow_squash_merge, bool), allow_squash_merge
+        assert allow_merge_commit is NotSet or isinstance(allow_merge_commit, bool), allow_merge_commit
+        assert allow_rebase_merge is NotSet or isinstance(allow_rebase_merge, bool), allow_rebase_merge
+        assert delete_branch_on_merge is NotSet or isinstance(delete_branch_on_merge, bool), delete_branch_on_merge
+        assert allow_update_branch is NotSet or isinstance(allow_update_branch, bool), allow_update_branch
```
Comment (EnricoMi) [inline github/Organization.py:466]: they all can use `is_optional`

File: github/Organization.py [hunk 6]
```diff
@@ -635,16 +527,16 @@ def create_secret(
                 isinstance(element, github.Repository.Repository) for element in selected_repositories
             ), selected_repositories
         else:
-            assert selected_repositories is github.GithubObject.NotSet
+            assert selected_repositories is NotSet
```
Comment (EnricoMi) [inline github/Organization.py:530]:
```suggestion
            assert is_undefined(selected_repositories)
```

File: github/Organization.py [hunk 7]
```diff
@@ -706,22 +598,18 @@ def create_team(
...
-        assert privacy is github.GithubObject.NotSet or isinstance(privacy, str), privacy
-        assert description is github.GithubObject.NotSet or isinstance(description, str), description
-        post_parameters = {
-            "name": name,
-        }
-        if repo_names is not github.GithubObject.NotSet:
+        assert is_optional_list(repo_names, github.Repository.Repository), repo_names
+        assert permission is NotSet or isinstance(permission, str), permission
+        assert privacy is NotSet or isinstance(privacy, str), privacy
+        assert description is NotSet or isinstance(description, str), description
```
Comment (EnricoMi) [inline github/Organization.py:604]:
```suggestion
        assert is_optional(permission, str), permission
        assert is_optional(privacy, str), privacy
        assert is_optional(description, str), description
```

File: github/Organization.py [hunk 8]
```diff
@@ -706,22 +598,18 @@ def create_team(
...
             post_parameters["repo_names"] = [element._identity for element in repo_names]
-        if permission is not github.GithubObject.NotSet:
+        if is_defined(permission):
             post_parameters["permission"] = permission
-        if privacy is not github.GithubObject.NotSet:
+        if is_defined(privacy):
             post_parameters["privacy"] = privacy
-        if description is not github.GithubObject.NotSet:
+        if is_defined(description):
             post_parameters["description"] = description
```
Comment (EnricoMi) [inline github/Organization.py:613]: should use `NotSet.remove_unset_items`

File: github/Organization.py [hunk 9]
```diff
@@ -749,14 +637,14 @@ def create_variable(
                 isinstance(element, github.Repository.Repository) for element in selected_repositories
             ), selected_repositories
         else:
-            assert selected_repositories is github.GithubObject.NotSet
+            assert selected_repositories is NotSet
```
Comment (EnricoMi) [inline github/Organization.py:640]:
```suggestion
            assert is_undefined(selected_repositories)
```

File: github/Organization.py [hunk 10]
```diff
@@ -812,58 +700,50 @@ def delete_hook(self, id):
...
-        assert location is github.GithubObject.NotSet or isinstance(location, str), location
-        assert name is github.GithubObject.NotSet or isinstance(name, str), name
+        assert billing_email is NotSet or isinstance(billing_email, str), billing_email
+        assert blog is NotSet or isinstance(blog, str), blog
+        assert company is NotSet or isinstance(company, str), company
+        assert description is NotSet or isinstance(description, str), description
+        assert email is NotSet or isinstance(email, str), email
+        assert location is NotSet or isinstance(location, str), location
+        assert name is NotSet or isinstance(name, str), name
         post_parameters = dict()
```
Comment (EnricoMi) [inline github/Organization.py:721]: should use `NotSet.remove_unset_items`

File: github/Organization.py [hunk 11]
```diff
@@ -812,58 +700,50 @@ def delete_hook(self, id):
...
-        assert email is github.GithubObject.NotSet or isinstance(email, str), email
-        assert location is github.GithubObject.NotSet or isinstance(location, str), location
-        assert name is github.GithubObject.NotSet or isinstance(name, str), name
+        assert billing_email is NotSet or isinstance(billing_email, str), billing_email
+        assert blog is NotSet or isinstance(blog, str), blog
+        assert company is NotSet or isinstance(company, str), company
+        assert description is NotSet or isinstance(description, str), description
+        assert email is NotSet or isinstance(email, str), email
+        assert location is NotSet or isinstance(location, str), location
+        assert name is NotSet or isinstance(name, str), name
```
Comment (EnricoMi) [inline github/Organization.py:720]:
```suggestion
        assert is_optional(billing_email, str), billing_email
        assert is_optional(blog, str), blog
        assert is_optional(company, str), company
        assert is_optional(description, str), description
        assert is_optional(email, str), email
        assert is_optional(location, str), location
        assert is_optional(name, str), name
```

File: github/Organization.py [hunk 12]
```diff
@@ -876,42 +756,39 @@ def edit_hook(
         assert isinstance(id, int), id
         assert isinstance(name, str), name
         assert isinstance(config, dict), config
-        assert events is github.GithubObject.NotSet or all(isinstance(element, str) for element in events), events
-        assert active is github.GithubObject.NotSet or isinstance(active, bool), active
-        post_parameters = {
+        assert is_optional_list(events, str), events
+        assert active is NotSet or isinstance(active, bool), active
```
Comment (EnricoMi) [inline github/Organization.py:760]:
```suggestion
        assert is_optional(active, bool), active
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]

**Comment by trim21** [comment]
this is strange, I didn't touch any test files but tests are broken

File: github/Organization.py [hunk 13]
```diff
@@ -1190,31 +1031,28 @@ def invite_user(
...
-        assert user is github.GithubObject.NotSet or isinstance(user, github.NamedUser.NamedUser), user
-        assert email is github.GithubObject.NotSet or isinstance(email, str), email
-        assert (email is github.GithubObject.NotSet) ^ (
-            user is github.GithubObject.NotSet
-        ), "specify only one of email or user"
-        parameters = {}
-        if user is not github.GithubObject.NotSet:
+        assert is_optional(user, github.NamedUser.NamedUser), user
+        assert is_optional(email, str), email
+        assert is_undefined(email) ^ is_undefined(user), "specify only one of email or user"
```
Comment (EnricoMi) [inline github/Organization.py:1036]: Operator `!=` might be more readable
```suggestion
        assert is_defined(email) != is_defined(user), "specify only one of email or user"
```

File: github/Organization.py [hunk 14]
```diff
@@ -462,30 +401,31 @@ def create_hook(
         """
         assert isinstance(name, str), name
         assert isinstance(config, dict), config
-        assert events is github.GithubObject.NotSet or all(isinstance(element, str) for element in events), events
-        assert active is github.GithubObject.NotSet or isinstance(active, bool), active
-        post_parameters = {
+        assert is_optional_list(events, str), events
+        assert is_optional(active, bool), active
+        post_parameters: dict[str, Any] = {
```
Comment (EnricoMi) [inline github/Organization.py:406]: why not having `NotSet.remove_unset_items` with `events` and `active` combined into this dict declaration?

File: github/Organization.py [hunk 15]
```diff
@@ -412,30 +356,25 @@ def create_fork(
...
-        :rtype: :class:`github.Repository.Repository`
         """
         assert isinstance(name, str), name
         assert isinstance(repo, github.Repository.Repository), repo
-        assert description is github.GithubObject.NotSet or isinstance(description, str), description
-        assert private is github.GithubObject.NotSet or isinstance(private, bool), private
-        post_parameters = {
+        assert is_optional(description, str), description
+        assert is_optional(private, bool), private
+        post_parameters: dict[str, Any] = {
```
Comment (EnricoMi) [inline github/Organization.py:371]: Why not using `NotSet.remove_unset_items` here?

File: github/Organization.py [hunk 16]
```diff
@@ -39,351 +39,299 @@
...
         """
         :calls: `PUT /orgs/{org}/memberships/{user} <https://docs.github.com/en/rest/reference/orgs#update-an-organization-membership-for-the-authenticated-user>`_
-        :param member: :class:`github.NamedUser.NamedUser`
-        :param role: string
-        :rtype: None
         """
-        assert role is github.GithubObject.NotSet or isinstance(role, str), role
+        assert is_optional(role, str), role
         assert isinstance(member, github.NamedUser.NamedUser), member
         put_parameters = {}
```
Comment (EnricoMi) [inline github/Organization.py:325]: candidate for `NotSet.remove_unset_items`

File: github/Organization.py [hunk 17]
```diff
@@ -945,121 +798,113 @@ def get_hook_deliveries(
...
-        if filter is not github.GithubObject.NotSet:
+        :rtype: :class:`PaginatedList` of :class:`github.Issue.Issue`
+        """
+        assert is_optional(filter, str), filter
+        assert is_optional(state, str), state
+        assert is_optional_list(labels, github.Label.Label), labels
+        assert is_optional(sort, str), sort
+        assert is_optional(direction, str), direction
+        assert is_optional(since, datetime), since
+        url_parameters: dict[str, Any] = {}
```
Comment (EnricoMi) [inline github/Organization.py:825]: candidate for `NotSet.remove_unset_items`

File: github/Organization.py [hunk 18]
```diff
@@ -945,121 +798,113 @@ def get_hook_deliveries(
...
 
-    def get_projects(self, state=github.GithubObject.NotSet):
+    def get_projects(self, state: Opt[str] = NotSet) -> PaginatedList[Project]:
         """
         :calls: `GET /orgs/{org}/projects <https://docs.github.com/en/rest/reference/projects#list-organization-projects>`_
-        :rtype: :class:`github.PaginatedList.PaginatedList` of :class:`github.Project.Project`
-        :param state: string
         """
 
         url_parameters = dict()
```
Comment (EnricoMi) [inline github/Organization.py:865]: candidate for `NotSet.remove_unset_items`

File: github/Organization.py [hunk 19]
```diff
@@ -945,121 +798,113 @@ def get_hook_deliveries(
...
         """
         :calls: `GET /orgs/{org}/outside_collaborators <https://docs.github.com/en/rest/reference/orgs#outside-collaborators>`_
         :param filter_: string
-        :rtype: :class:`github.PaginatedList.PaginatedList` of :class:`github.NamedUser.NamedUser`
+        :rtype: :class:`PaginatedList` of :class:`github.NamedUser.NamedUser`
         """
-        assert filter_ is github.GithubObject.NotSet or isinstance(filter_, str), filter_
+        assert is_optional(filter_, str), filter_
 
         url_parameters = {}
```
Comment (EnricoMi) [inline github/Organization.py:897]: candidate for `NotSet.remove_unset_items`

File: github/Organization.py [hunk 20]
```diff
@@ -1292,15 +1122,18 @@ def create_migration(
         """
         assert isinstance(repos, (list, tuple)), repos
         assert all(isinstance(repo, str) for repo in repos), repos
-        assert lock_repositories is github.GithubObject.NotSet or isinstance(lock_repositories, bool), lock_repositories
-        assert exclude_attachments is github.GithubObject.NotSet or isinstance(
-            exclude_attachments, bool
-        ), exclude_attachments
+        assert is_optional(lock_repositories, bool), lock_repositories
+        assert is_optional(exclude_attachments, bool), exclude_attachments
         post_parameters = {"repositories": repos}
```
Comment (EnricoMi) [inline github/Organization.py:1127]: Why not merging the `update` from below into this?
```suggestion
        post_parameters = NotSet.remove_unset_items(
                {
                    "repositories": repos,
                    "lock_repositories": lock_repositories,
                    "exclude_attachments": exclude_attachments,
                }
            )
```

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**